src/context/walletConnectContext.tsx

```diff
@@ -0,0 +1,329 @@
+            .map((namespace) => namespace.chains)
+            .flat() as string[]
+
+          web3Modal.openModal({ uri, standaloneChains })
+        }
+
+        //TODO: if user closes modal before approving on mobile state is not resetted
+
+        session = await approval()
+        console.log("Established session:", session)
+
+        setSession(session)
+
+        web3Modal.closeModal()
+
+        await onSuccess(session)
+      } catch (e) {
+        web3Modal.closeModal()
+        onError?.(e)
+
+        throw e
+      }
+    },
+    [client]
+  )
+
+  const identifyUser = useCallback(
+    async (network: Network, session: SessionTypes.Struct) => {
+      if (typeof client === "undefined") {
+        throw new Error("WalletConnect is not initialized")
+      }
+      if (typeof session === "undefined") {
+        throw new Error("Session is not connected")
+      }
+
+      try {
+        const message: Connex.Vendor.CertMessage = {
+          purpose: "identification",
+          payload: {
+            type: "text",
+            content: "Sign a certificate to prove your identity",
+          },
+        }
+
+        const result: Connex.Vendor.CertResponse = await client.request({
+          topic: session.topic,
+          chainId: `vechain:${network}`,
+          request: {
+            method: DEFAULT_METHODS.IDENTIFY,
+            params: [message],
+          },
+        })
+
+        const cert: Certificate = {
+          purpose: message.purpose,
+          payload: message.payload,
+          domain: result.annex.domain,
+          timestamp: result.annex.timestamp,
+          signer: result.annex.signer,
+          signature: result.signature,
+        }
+
+        console.log("Signed cert", cert)
+        Certificate.verify(cert)
+        console.log("Cert verified")
+
+        return cert
+      } catch (error) {
+        console.error("SignClient.identifyUser failed:", error)
+
+        // Disconnect from session if user rejects identify signature request
+        await client.disconnect({
+          topic: session.topic,
+          reason: getSdkError("USER_DISCONNECTED"),
+        })
+
+        throw error
+      }
+    },
+    [client, session]
+  )
+
+  const disconnect = useCallback(async () => {
+    if (typeof client === "undefined") {
+      throw new Error("WalletConnect client is not initialized")
+    }
+    if (typeof session === "undefined") {
+      throw new Error("Session is not connected")
+    }
+
+    try {
+      await client.disconnect({
+        topic: session.topic,
+        reason: getSdkError("USER_DISCONNECTED"),
+      })
+    } catch (error) {
+      console.error("SignClient.disconnect failed:", error)
+    } finally {
+      // Reset app state after disconnect.
+      reset()
+    }
+  }, [client, session])
+
+  const subscribeToEvents = useCallback(async (_client: Client) => {
+    if (typeof _client === "undefined") {
+      throw new Error("WalletConnect is not initialized")
+    }
```

Comment:
See above. Do we really need to pass the client here ? We should already have it in the context